ast/desugar/Desugar.cc

```diff
@@ -301,7 +301,8 @@ ExpressionPtr validateRBIBody(DesugarContext dctx, ExpressionPtr body) {
 }
 
 ExpressionPtr buildMethod(DesugarContext dctx, core::LocOffsets loc, core::LocOffsets declLoc, core::NameRef name,
-                          unique_ptr<parser::Node> &argnode, unique_ptr<parser::Node> &body, bool isSelf) {
+                          unique_ptr<parser::Node> &argnode, unique_ptr<parser::Node> &body, bool isSelf,
+                          bool removeBody) {
```

Comment:
`removeBody` feels misplaced on the method signature for `buildMethod`. Seems like this should be a property of the DesugarContext to me. Defer to @jez since I don't have a ton of context on this step.

Reply:
Mm ok. Technically I can do the `dctx.ctx...` check here (which basically uses `DesugarContext`), instead of in the caller.